### ISSUE TITLE
FAM file monitroring system File integrity HIDS to syslog and to local GUI from suricata.

### ISSUE THREAD

**Issue by orwellencarnation** [Librerouter/Librekernel#262]
Implement this sending syslog to 10.0.0.200 and to local syslog server if when is placed

ossec hids , inotify, fschange, aide,Samhain, Tripwire open,

    FAM (File Alteration Monitor) – It is one of the oldest portable event monitors. It sends events to an application when changes are made to files or directories that an application has registered to be monitored. It is also complicated.
    Gamin – newer and simpler than FAM. It tries to be compatible with FAM while not implementing many of the obscure feature. It is moderately maintained and widely available on many distros. It is portable, but the focus of development and testing is on Linux. It has BSD support and can be found in FreeBSD Ports.
    dnotify – introduced in 2.4 kernel series. It can only watch directories and requires maintaining an open file descriptor to the directory that the user wants to watch. It has been obsoleted by inotify.

**Comment by orwellencarnation** [comment]
Nguyen you need to put your questions in the issue if you dont understand what to do

**Comment by nguyenhoangtran11** [comment]
Do we need to implement log rotation in the server?

**Comment by nguyenhoangtran11** [comment]
The differences between FAM and HIDS:
+ FAM (File Alteration Monitor) allows applications to watch certain files and be notified when they are modified. This greatly aids the applications, because before FAM existed, such applications would have to read the disk repeatedly to detect any changes, which resulted in high disk and CPU usage. 

+ HIDS (Host-based Intrusion Detection System) is an intrusion detection system that monitors and analyzes the internals of a computing system as well as  the network packets on its network interfaces. It will monitor all or parts of the dynamic behavior and the state of a computer system. Besides, HIDS might detect which program accesses what resources and discover that (for example, a word-processor has suddenly and inexplicably started modifying the system password database). Moreover, HIDS might look at the state of a system, its stored information, whether in RAM, in the file system, log files or elsewhere; and check that the contents of these appear as expected.

**Comment by nguyenhoangtran11** [comment]
The FAM system consists of two parts:
+ famd — the FAM Daemon, which provides notifications and listens for requests. Administrators can configure it by editing the file /etc/fam.conf
+ libfam — the interface to the client

**Comment by orwellencarnation** [comment]
We need lighter FAM and the lighter HIDS as possible.
Both are going to syslog to an external syslog in 10.0.0.200

**Comment by nguyenhoangtran11** [comment]
+ ossec hids (http://ossec.github.io/downloads.html): The current release version is 2.9.0.  It is a scalable, multi-platform, open source Host-based Intrusion Detection System (HIDS). It has a powerful correlation and analysis engine, integrating log analysis, file integrity checking, Windows registry monitoring, centralized policy enforcement, rootkit detection, real-time alerting and active response.It runs on most operating systems, including Linux, OpenBSD, FreeBSD, MacOS, Solaris and Windows. It can support sending alerts to one or more syslog servers. It is a free software and will remain so in the future. The code size is small. It is around 1.6MB.

**Comment by orwellencarnation** [comment]
We need just HIDS client not server as the server will be 10.0.0.200 in other machine.
So HIDS or FAM FIM of file integrity system would send all syslogs to 10.0.0.200 

for FIM  FAM which one you would select?

**Comment by nguyenhoangtran11** [comment]
Thank for your info. http://aide.sourceforge.net/ is better. I will use it

For FIM FAM, I think we will use inotify. It is available on Debian GNU/Linux | Ubuntu. We can get it via sudo apt-get install inotify-tools

**Comment by orwellencarnation** [comment]
why is better, please when you decide tell me why no just yes

**Comment by nguyenhoangtran11** [comment]
aide is better because it is lighter than ossec. Moreover, it is available on Debian GNU/Linux | Ubuntu. We can get it via: sudo apt-get install aide

**Comment by orwellencarnation** [comment]
Would you plan the changes you purpose for app instalaltion script and app configration script for AIDE and INotify regarding the requirements?

**Comment by orwellencarnation** [comment]
Did you installaed your lab?

**Comment by nguyenhoangtran11** [comment]
Which files/directories need to be checked and notified when there is a change?
How to  check whether the log is written to 10.0.0.200 server? eg: how to access it?

**Comment by orwellencarnation** [comment]
All

**Comment by nguyenhoangtran11** [comment]
I don't think we will monitor all files/directories on the system as we can get a lot of "false" log on the server 10.0.0.200. Moreover, if we do such thing, the log is very very big and it will occupy all spaces on hard disk soon. I suggested we should check and ignore following directories:

Things to be monitored:
 + /etc
 + /bin         
 + /sbin    
 + /usr/bin   
 + /usr/sbin 
 + /usr/local/bin
 + /usr/local/sbin 
 + /usr/games    
 + /lib      
 + /usr/lib   
 + /usr/local/lib 

# Things to be ignored as they change too often
/dev
/proc
/mnt
/usr/src
/usr/doc
/usr/share/doc
/var/log/.*     
/var/spool/.*   
/var/adm/utmp*

if any directory is missed, please add more.

Another thing, how long do we need to check the change for all registered files/directories automatically? 1 hour or 2 hours or 1 day, 1 week, etc.

**Comment by orwellencarnation** [comment]
I dont want to know which files has being accesed 
I want to know which files has being changed in the files system and how often
And the both system FAM and HIDS should not send single events but sumarized events

**Comment by nguyenhoangtran11** [comment]
Another tool that can be used here: https://rflament.github.io/loggedfs/

**Comment by nguyenhoangtran11** [comment]
Support native syslog:
+ LoggedFS




No support native syslog:
+ inotify
+ dnotify: less feature than inotify and all feature of dnotify are included on inotify
+ fanotify: similar to inotify but no support for create, delete, and move events
+ fschange
+ aide

**Comment by orwellencarnation** [comment]
What about OSSEC?

**Comment by orwellencarnation** [comment]
What about Gamin?

**Comment by nguyenhoangtran11** [comment]
I update them in previous comment. What is our choice now? I suggested to use inotify as it's a Linux kernel module. Moreover, it's fast and lightweight and should be available for every Linux distro flavor

**Comment by orwellencarnation** [comment]
Install and configure 
    LoggedFS
    OSSEC
    Samhain

Create 2 scripts

a ) Installation
b) Configuration
